app/controllers/pages_controller.rb

```diff
@@ -0,0 +1,76 @@
+class PagesController < ApplicationController
+  before_action :require_auth, only: [:new, :create, :show, :edit, :update, :destroy]
+  before_action :set_page, only: [:edit, :update, :destroy ]
+
+  # ------------------------
+  # public methods (no auth)
+  # ------------------------
+
+  # GET /pages
+  def index
+    @pages = Page.all
+  end
+
+  # GET /pages/code/view
+  def view
+    @page = Page.find_by code: params[:id]
```

Comment:
I think we should consider renaming `code` to `slug` to follow the [Semantic URL Slug definition](https://en.wikipedia.org/wiki/Semantic_URL#Slug) 

I know it's called `code` in the existing system, but this seems like a good opportunity to give that property a more accurate name?